### PR TITLE
Fix transcendental secant plane lemmas

### PR DESCRIPTION
**Pull request by nafur**
The refactoring of the transcendental solver introduced a subtle issue that lead to incorrect secant plane lemmas.
This PR fixes this issue, so that we now produce the correct lemmas again.

### CODE REVIEW

File: src/theory/arith/nl/transcendental/transcendental_state.cpp [hunk 1]
```diff
@@ -294,12 +294,12 @@ std::pair<Node, Node> TranscendentalState::getClosestSecantPoints(TNode e,
...
 
 Node TranscendentalState::mkSecantPlane(
-    TNode arg, TNode b, TNode c, TNode approx, TNode approx_c)
+    TNode arg, TNode b, TNode c, TNode approx_b, TNode approx_c)
 {
   NodeManager* nm = NodeManager::currentNM();
   // Figure 3 : P(l), P(u), for s = 0,1
-  Node approx_b =
-      Rewriter::rewrite(approx.substitute(d_taylor.getTaylorVariable(), b));
+  // Node approx_b =
```
Comment (ajreynol) [inline src/theory/arith/nl/transcendental/transcendental_state.cpp:301]: Is this really not needed? If so delete.

File: src/theory/arith/nl/transcendental/transcendental_state.h [hunk 2]
```diff
@@ -89,119 +89,120 @@ struct TranscendentalState
...
-    Node d_pi_2;
-    /** -PI/2 */
-    Node d_pi_neg_2;
-    /** -PI */
-    Node d_pi_neg;
-    /** the concrete lower and upper bounds for PI */
-    Node d_pi_bound[2];
+                      int concavity);
+
+  Node d_true;
```
Comment (ajreynol) [inline src/theory/arith/nl/transcendental/transcendental_state.h:107]: May be a formatting issue here.

File: src/theory/arith/nl/transcendental/transcendental_state.cpp (hunk 1 above)
Comment (nafur) [reply src/theory/arith/nl/transcendental/transcendental_state.cpp]: Did so, and moved the comment to where it now belongs.

File: src/theory/arith/nl/transcendental/transcendental_state.h (hunk 2 above)
Comment (nafur) [reply src/theory/arith/nl/transcendental/transcendental_state.h:107]: Yes, but the new formatting is the right one.

**Review by ajreynol — APPROVED** [review]

**Pull request merged**